integration/test/createnodepool/createnodepool_test.go

```diff
@@ -0,0 +1,234 @@
+			corev1.AddToScheme,
+		}
+		err = appSchemeBuilder.AddToScheme(runtimeScheme)
+		if err != nil {
+			t.Fatal(err)
+		}
+		appTest, err = apptest.New(apptest.Config{
+			KubeConfigPath: env.KubeConfig(),
+			Logger:         logger,
+			Scheme:         runtimeScheme,
+		})
+		if err != nil {
+			t.Fatal(err)
+		}
+	}
+
+	{
+		err = appTest.EnsureCRDs(ctx, getRequiredCRDs())
+		if err != nil {
+			t.Fatal(err)
+		}
+	}
+
+	{
+		valuesYAML, err := values.YAML(env.AzureClientID(), env.AzureClientSecret(), env.AzureTenantID(), env.AzureSubscriptionID())
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		apps := []apptest.App{
+			{
+				CatalogName:   prodCatalogName,
+				Name:          "cert-manager-app",
+				Namespace:     metav1.NamespaceDefault,
+				Version:       "2.3.1",
+				WaitForDeploy: true,
+			},
+			{
+				CatalogName:   testCatalogName,
+				Name:          "azure-admission-controller",
+				Namespace:     metav1.NamespaceDefault,
+				SHA:           env.CircleSHA(),
+				ValuesYAML:    valuesYAML,
+				WaitForDeploy: true,
+			},
+		}
+		err = appTest.InstallApps(ctx, apps)
+		if err != nil {
+			t.Fatal(err)
+		}
+	}
+
+	err = ensureCRsExist(appTest.CtrlClient(), []string{"namespaces.yaml", "organization.yaml", "azurecluster.yaml", "cluster.yaml", "azuremachinepool.yaml", "machinepool.yaml"})
+	if err != nil {
+		deleteCRs(appTest.CtrlClient(), []string{"namespaces.yaml", "organization.yaml", "azurecluster.yaml", "cluster.yaml", "azuremachinepool.yaml", "machinepool.yaml"})
+		t.Fatal(err)
+	}
+}
+
+func getRequiredCRDs() []*apiextensionsv1.CustomResourceDefinition {
+	clusterCRD := crd.LoadV1(clusterAPIGroup, "Cluster")
+	azureClusterCRD := crd.LoadV1(infrastructureAPIGroup, "AzureCluster")
+	azureMachineCRD := crd.LoadV1(infrastructureAPIGroup, "AzureMachine")
+	machinePoolCRD := crd.LoadV1(experimentalClusterAPIGroup, "MachinePool")
+	azureMachinePoolCRD := crd.LoadV1(experimentalInfrastructureAPIGroup, "AzureMachinePool")
+	organizationCRD := crd.LoadV1(securityAPIGroup, "Organization")
+
+	return []*apiextensionsv1.CustomResourceDefinition{
+		corev1alpha1.NewAzureClusterConfigCRD(),
+		corev1alpha1.NewSparkCRD(),
+		providerv1alpha1.NewAzureConfigCRD(),
+
+		clusterCRD,
+		azureClusterCRD,
+		azureMachineCRD,
+		machinePoolCRD,
+		azureMachinePoolCRD,
+		organizationCRD,
+	}
+}
+
+func ensureCRsExist(client client.Client, inputFiles []string) error {
+	for _, f := range inputFiles {
+		o, err := loadCR(f)
+		if err != nil {
+			return err
+		}
+
+		err = client.Create(context.Background(), o)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func deleteCRs(client client.Client, inputFiles []string) {
+	for _, f := range inputFiles {
+		o, err := loadCR(f)
+		if err != nil {
+			panic(err.Error())
+		}
+
+		err = client.Delete(context.Background(), o)
+		if apierrors.IsNotFound(err) {
+			// Ok
+		} else if err != nil {
+			panic(err.Error())
+		}
+	}
+}
+
+func loadCR(fName string) (runtime.Object, error) {
```

Comment:
Would it make sense to figure out a package under `/pkg` and put these generic functions there? We have this now in quite many places?

Maybe we put a line to that refactoring task issue and take care of it separately (no need to context switch here)?